azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -117,17 +143,55 @@ public int createImageType(ImageType imageType) {
 
   @Override
   public List<ImageType> getAllImageTypes() throws ImageMgmtException {
-    FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
+    final FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
     try {
-      return databaseOperator
+      return this.databaseOperator
           .query(FetchImageTypeHandler.FETCH_ALL_IMAGE_TYPES, fetchImageTypeHandler, true);
-    } catch (SQLException ex) {
+    } catch (final SQLException ex) {
       log.error(FetchImageTypeHandler.FETCH_ALL_IMAGE_TYPES + " failed.", ex);
       throw new ImageMgmtDaoException(
           "Unable to fetch all image type metadata ");
     }
   }
 
+  @Override
+  public List<ImageType> getAllImageTypesWithOwnerships() throws ImageMgmtException {
+    final FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
+    List<ImageType> imageTypes = new ArrayList<>();
+    try {
+      imageTypes = this.databaseOperator
+          .query(FetchImageTypeHandler.FETCH_ALL_IMAGE_TYPES, fetchImageTypeHandler, true);
+      for (final ImageType imageType : imageTypes) {
+        imageType.setOwnerships(getImageOwnerships(imageType.getName()));
```

Comment:
You are executing many queries over here. Can you please execute single query with joins and prepare List of image types in handler?

Reply:
Added single query to get the owners based on all the image types.